core/src/main/java/org/apache/iceberg/hadoop/Util.java

```diff
@@ -84,10 +88,38 @@ public static String[] blockLocations(FileIO io, ScanTaskGroup<?> taskGroup) {
     return locations.toArray(HadoopInputFile.NO_LOCATION_PREFERENCE);
   }
 
+  public static boolean usesHadoopFileIO(FileIO io, String location) {
```

Comment:
I think this can be private and we can simply use `mayHaveBlockLocations` in Spark.